src/System.Runtime.Extensions/src/System/Environment.Unix.cs

```diff
@@ -0,0 +1,205 @@
+                    if (!string.IsNullOrEmpty(home))
+                    {
+                        return home;
+                    }
+                    break;
+
+                // TODO: Add more special folder handling
+            }
+
+            throw new PlatformNotSupportedException();
+        }
+
+        public static string[] GetLogicalDrives() => Interop.Sys.GetAllMountPoints().ToArray();
+
+        private static bool Is64BitOperatingSystemWhen32BitProcess => false;
+
+        public static string MachineName => Interop.Sys.GetHostName();
+
+        public static string NewLine => "\n";
+
+        private static Lazy<OperatingSystem> s_osVersion = new Lazy<OperatingSystem>(() =>
+        {
+            int major = 0, minor = 0, build = 0, revision = 0;
+
+            // Get the uname's utsname.release.  Then parse it for the first four numbers found.
+            // This isn't perfect, but Version already doesn't map exactly to all possible release
+            // formats, e.g. 
+            string release = Interop.Sys.GetUnixRelease();
+            if (release != null)
+            {
+                int i = 0;
+                major = FindAndParseNextNumber(release, ref i);
+                minor = FindAndParseNextNumber(release, ref i);
+                build = FindAndParseNextNumber(release, ref i);
+                revision = FindAndParseNextNumber(release, ref i);
+            }
+
+            return new OperatingSystem(PlatformID.Unix, new Version(major, minor, build, revision));
+        });
+
+        private static int FindAndParseNextNumber(string text, ref int pos)
+        {
+            // Move to the beginning of the number
+            for (; pos < text.Length; pos++)
+            {
+                char c = text[pos];
+                if ('0' <= c && c <= '9')
+                {
+                    break;
+                }
+            }
+
+            // Parse the number;
+            int num = 0;
+            for (; pos < text.Length; pos++)
+            {
+                char c = text[pos];
+                if ('0' <= c && c <= '9')
+                {
+                    num = (num * 10) + (c - '0');
+                }
+                else break;
+            }
+            return num;
+        }
+
+        public static int ProcessorCount => (int)Interop.Sys.SysConf(Interop.Sys.SysConfName._SC_NPROCESSORS_ONLN);
+
+        public static string SystemDirectory => GetFolderPathCore(SpecialFolder.System, SpecialFolderOption.None);
+
+        public static int SystemPageSize => (int)Interop.Sys.SysConf(Interop.Sys.SysConfName._SC_PAGESIZE);
+
+        public static int TickCount
+        {
+            get
+            {
+                // TODO: While this is functional, it would be better performing to put an implementation into System.Native
+                // similar to what's used in the libcoreclr's GetTickCount() implementation.
```

Comment:
`Environment.TickCount` tends to be used on performance sensitive paths because of it has always been the lowest overhead way to get a time stamp. We should keep an eye on perf with this rewrite - on both Windows and Unix.